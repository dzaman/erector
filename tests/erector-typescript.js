const child_process = require('child_process');

describe('erector-typescript', () => {
  // Only run this test in CI because it's super slow
  // Also output snapshot is directory specific -- though that could be changed
  const argv = process.env.npm_config_argv ? JSON.parse(process.env.npm_config_argv) : { remain: [] };
  const test_fn = argv.remain.includes('--ci') ? test : test.skip;

  test_fn('typescript compilation of consumer', () => {
    const result = child_process.spawnSync('node_modules/.bin/tsc', [
      './tests/fixtures/erector-consumer.ts',
      '--noEmit',
      '--esModuleInterop',
    ]);
    expect(result.stdout.toString()).toMatchSnapshot();
  });
});
