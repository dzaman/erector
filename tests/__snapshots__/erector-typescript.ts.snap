// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`erector-typescript typescript compilation of consumer 1`] = `
"tests/fixtures/erector-consumer.ts(23,16): error TS2339: Property 'does_not_exist' does not exist on type 'typeof import(\\"/Users/matt/dev/erector/lib/erector\\")'.
tests/fixtures/erector-consumer.ts(41,24): error TS2339: Property 'does_not_exist' does not exist on type 'typeof erector'.
"
`;
